models/src/it/java/io/sphere/sdk/producttypes/ProductTypeIntegrationTest.java

```diff
@@ -75,6 +76,35 @@ public void booleanAttribute() throws Exception {
     }
 
     @Test
+    public void nestedAttribute() throws Exception {
+        TestCleaner.withCleaner(cleaner -> {
```

Comment:
I think it is not a good idea to introduce different testing styles, the TestCleaner is in competition with the different test fixture classes and they are not compatible with each other.

The Fixture classes are well, because they enable cleanup before the use, thats cool in case of crashes and also they hide how to delete data. With the TestCleaner you duplicate how to delete something.

 I would accept this pull request with the TestCleaner but remove it myself in master. Further pull requests with a TestCleaner will not be accepted. I don't different architectures for the same thing in one product. It should all look alike and use few concepts. If we use a lot of concepts, it is required for contributors to learn each concept before starting and I actually want to make it easy to contribute.